src/isar_turtlebot/robotinterface.py

```diff
@@ -262,27 +248,21 @@ def _publish_navigation_task(self, pose: Pose) -> None:
         }
 
         self.bridge.execute_task.publish(message=pose_message)
-
-    def _get_run_id(self) -> Optional[str]:
-        status_msg: dict = self.bridge.mission_status.get_value()
-        try:
-            run_id: str = status_msg["status_list"][0]["goal_id"]["id"]
-            return run_id.replace("move_base-", "").split(".")[0].replace("-", "")
-        except (KeyError, IndexError):
-            self.logger.info("Failed to get current mission_id returning None")
-            return None
-
-    def _wait_for_updated_task(self, previous_run_id: str, timeout: int = 20) -> str:
         start_time: float = time.time()
-        current_run_id: str = self._get_run_id()
 
-        while current_run_id == previous_run_id:
+        while self._get_turtlebot_id() == self.id:
             time.sleep(0.1)
             execution_time: float = time.time() - start_time
-            if execution_time > timeout:
-                raise TimeoutError(
-                    f"Scheduling of task for TurtleBot3 timed out. Run ID: {current_run_id}"
-                )
-            current_run_id = self._get_run_id()
 
-        return current_run_id
+            if execution_time > self.inspection_task_timeout:
+                self.inspection_status = TurtlebotStatus.Failure
+                raise TimeoutError(f"Storing image for TurtleBot3 timed out.")
+
+    def _get_turtlebot_id(self) -> Optional[str]:
```

Comment:
Change to `_get_turtlebot_task_id(self)` to make it  clear that we are getting the id of the navigation task on the robot?

Reply:
This is a good catch, agree it is not clear which id this is. I renamed it to `self.goal_id`, as it is not the same as `task.id`. Other suggestions are welcome.